### PR TITLE
Push docker image to the OCP|OKD4 and deploy operator

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

File: build/push-okd4.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  validateROOT
+  sudo -E mkdir -p "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}"
+fi
+
+if [ -f "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}/tls.crt" ]; then
+  echo "Instance file detected"
+else
+  echo "Instance file not found, extracting..."
+  validateROOT
+  sudo -E oc extract secret/router-ca --keys=tls.crt -n openshift-ingress-operator --to="${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}"
```
Comment (rigazilla) [inline build/push-okd4.sh]: I have oc installed under $HOME/bin and it's not found by sudo:
```
$ sudo -E oc
sudo: oc: command not found
```
I think is quite common to have oc not in the default path, how about change to:
```
OC=$(which oc)
sudo -E $OC
```
?

File: build/push-okd4.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+validateOC() {
+  OC_USER_NAME=$(oc whoami 2>/dev/null)
```
Comment (rigazilla) [inline build/push-okd4.sh]: Strange problem here for me:
```
$ oc whoami 
kube:admin
```
but 
```
docker login -u kube:admin
```
fails and
```
docker login -u kubeadmin
```
works

File: build/push-okd4.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+if [ -f "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}/tls.crt" ]; then
+  echo "Instance file detected"
+else
+  echo "Instance file not found, extracting..."
+  validateROOT
+  sudo -E oc extract secret/router-ca --keys=tls.crt -n openshift-ingress-operator --to="${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}"
+fi
+
+docker login -u "$OC_USER_NAME" -p "${OC_USER_TOKEN}" "${REGISTRY_ROUTE}"
```
Comment (Crumby) [inline build/push-okd4.sh]: `oc registry login --skip-check` does the same job once you are logged in and you can drop the env vars.

Comment (dmvolod) [reply build/push-okd4.sh]: Thanks, @Crumby will update
It'a strange, that out KB note related to this problem, doesn't know about this option

File: build/push-okd4.sh (hunk 1 above)
Comment (dmvolod) [reply build/push-okd4.sh]: @rigazilla, updated without sudo with solution provided by @Crumby

Comment (dmvolod) [reply build/push-okd4.sh]: Yes, thanks for suggestion, will update

File: build/push-okd4.sh (hunk 2 above)
Comment (dmvolod) [reply build/push-okd4.sh]: @rigazilla will update with solution provided by Pavel

File: build/push-okd4.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  fi
+}
+
+validateOC
+
+DOCKER_CERTS_FOLDER="/etc/docker/certs.d/"
+REGISTRY_ROUTE=$(oc get route default-route -n openshift-image-registry -o json | jq -r '.spec.host')
+OC_PROJECT_NAME=$(oc project -q)
+PROJECT_NAME=${PROJECT_NAME-$OC_PROJECT_NAME}
+VERSION=$(git describe --tags --always --dirty)
```
Comment (rigazilla) [inline build/push-okd4.sh]: Can we make it configurable as in build.sh?
`VERSION=${RELEASE_NAME-$(git describe --tags --always --dirty)}`

Comment (dmvolod) [reply build/push-okd4.sh]: Thanks, done

**Pull request merged**